xGame/client/src/Components/MatchSelect/MatchSelect.js

```diff
@@ -1,18 +1,73 @@
 import React from "react";
 import { Collapse } from "reactstrap";
 import PropTypes from "prop-types";
-import { Button } from '@material-ui/core';
+import { Button, Paper, Container } from '@material-ui/core';
+import { List, ListItem, ListItemSecondaryAction, ListItemText } from '@material-ui/core';
 import MatchCreateDialog from './MatchCreateDialog';
+import IconButton from '@material-ui/core/IconButton';
+import PlayIcon from '@material-ui/icons/PlayCircleOutline';
+import { useEffect, useState } from "react";
+import { Alert } from "@material-ui/lab";
+import { Formik } from "formik";
+
+import * as matchService from "../../service/matchService";
 
 const MatchSelect = (props) => {
-  const { isOpen, activeUser } = props;
+  const { isOpen, activeUser, setActiveMatch } = props;
   const [showMatchCreation, setShowMatchCreation] = React.useState(false);
+  const [selectError, setSelectError] = useState("");
+  var [matchList, setMatchList] = useState([]);
 
+  useEffect(() => {
+    async function fetchData() {
+      setMatchList(await matchService.getActiveMatches(activeUser?.id));
+      if (matchList?.error) setSelectError(matchList?.message);
+    }
+    fetchData();
+  }, [activeUser]);
 
   return (
     <Collapse isOpen={isOpen}>
       <div>
-        <p>add match selection data</p>
+        <br />
+        {selectError && (
+          <Alert
+            onClose={() => {
+              setSelectError("");
+            }}
+            severity="error"
+          >
+            {selectError}
+          </Alert>
+        )}
+        <Container maxWidth="sm">
+          <Paper elevation={0} style={{maxHeight: 200, overflow: 'auto'}}>
+            <List dense={true}>
+              {matchList?.map((X) => {
```

Comment:
matchList occasionally gets set to false, so you need to check for that here by just adding "matchList &&" like so:
`{matchList && matchList?.map((X) => {`